Source/AliveLibCommon/audio/MidiPlayer.hpp

```diff
@@ -0,0 +1,297 @@
+    };
+    struct AoVag
+    {
+        u32 iSize;
+        u32 iSampleRate;
+        std::vector<unsigned char> iSampleData;
+    };
+
+    struct ProgAtr final
+    {
+        u8 field_0_num_tones;
+        s8 field_1_vol;
+        s8 field_2_priority;
+        s8 field_3_mode;
+        s8 field_4_pan;
+        s8 field_5_reserved0;
+        s16 field_6_attr;
+        s32 field_8_reserved1;
+        s32 field_C_reserved2;
+    };
+
+    struct VabHeader final
+    {
+        s32 field_0_form;
+        s32 field_4_version;
+        s32 field_8_id;
+        s32 field_C_file_size;
+        s16 field_10_reserved0;
+        s16 field_12_num_progs;
+        s16 field_14_num_tones;
+        u16 field_16_num_vags;
+        s8 field_18_master_vol;
+        s8 field_19_master_pan;
+        s8 field_1A_attr1;
+        s8 field_1B_attr2;
+        s32 field_1C_reserved1;
+        ProgAtr field_20_progs[128];
+    };
+
+    struct SoundBlockInfo final
+    {
+        char_type* header_name;
+        char_type* body_name;
+        s32 vab_id;
+        u8* VabHeader;
+    };
+
+    struct SeqHeader
+    {
+        Uint32 mMagic;   // SEQp
+        Uint32 mVersion; // Seems to always be 1
+        Uint8 mResolutionOfQuaterNote[2];
+        Uint8 mTempo[3];
+        Uint8 mTimeSignatureBars;
+        Uint8 mTimeSignatureBeats;
+    };
+
+    struct SeqInfo
+    {
+        Uint32 iLastTime = 0;
+        Sint32 iNumTimesToLoop = 0;
+        Uint8 running_status = 0;
+    };
+
+    struct OpenSeqHandle final
+    {
+        char_type* mBsqName;
+        s32 generated_res_id; // A hash of the named which matches the resource Id
+        s8 sound_block_idx;
+        s8 volume;
+        s16 seqOpenId;
+        u8* ppSeq_Data;
+    };
+
+    struct SfxDefinition final
+    {
+        s8 block_idx;
+        s8 program;
+        s8 note;
+        s8 volume;
+        s16 pitch_min;
+        s16 pitch_max;
+    };
+
+    struct Vag final
+    {
+        u16 adsr_attack;
+        u16 adsr_sustain_level;
+        u16 adsr_decay;
+        u16 adsr_release;
+        u8 min;
+        u8 max;
+        s16 shift_cen;
+        u8 C;
+        u8 vol;
+        u8 priority;
+        u8 prog;
+        u8 vag;
+        s8 pad;
+    };
+
+    /*
+        Represents some data read from disk
+    */
+    struct ResourceData final
+    {
+        u8* data;
+        u32 size;
+        s32 optionalHash;
+    };
+
+    /*
+        Provides a way to read files from a "psx" disk.
+    */
+    class ResourceProvider
+    {
+    public:
+        /* Read a file within the game, example a .VB file */
+        virtual ResourceData* readFile(char_type* name)
+        {
+            name;
+            throw std::invalid_argument("readFile is not overridden");
+        }
+        /* AO reads sequence inline with data. AE reads from a sounds.dat file */
+        virtual ResourceData* readSeq(const char_type* fileName, const char_type* sequenceName)
+        { 
+            fileName;
```

Comment:
Probably safe to remove, like the other lines like this in this file (and others)